pennylane_lightning/lightning_qubit.py

```diff
@@ -220,10 +220,11 @@ def adjoint_jacobian(self, tape, starting_state=None, use_device_state=False):
 
         ops_serialized = adj.create_ops_list(*ops_serialized)
 
+        tape_trainable_params_set = set(tape.trainable_params)
```

Comment:
@maliasadi just something to keep in mind is that the reason we changed `trainable_params` from being a set is that we were affected by a rare python bug where set order of integers is not guaranteed to be stable!

So something to be careful with if you plan to iterate over a set, and expect the integers to be ordered.

Reply:
I think it should be fine --- we don't really care about order as much as the existence of indices. We iterate over them in the order given to CPP, and assume no guarantees on being sorted (I think).

Reply:
Thank you @josh146 @mlxd! I update `AdjointDiff` method to work with `trainable_params` as a list.